### PR TITLE
ddl: fix a bug in AlterSchema job and add more tests

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve?
This PR fixes 2 potential bugs in AlterSchema DDL job

### What is changed and how it works?
The change is simple enough, please check the codes.

### Check List 

Tests 

 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @zimulala @winkyao @crazycs520

File: ddl/schema_test.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ package ddl
 
 import (
 	"context"
+	"github.com/pingcap/errors"
```
Comment (crazycs520) [inline ddl/schema_test.go]: Put this in right line.

Comment (bb7133) [reply ddl/schema_test.go]: thanks!

**Pull request merged**